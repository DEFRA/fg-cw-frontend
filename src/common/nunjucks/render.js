import camelCase from "lodash/camelCase.js";
import { environment } from "./nunjucks.js";

/**
 * @param {string} name
 * @param {object} params
 * @param {string} [block]
 */
export const render = (name, params, block) => {
  const macro = {
    path: `${name}/macro.njk`,
    name: camelCase(name.split("-")),
    params: JSON.stringify(params, null, 2),
  };

  const callComponent = block
    ? `{%- call ${macro.name}(${macro.params}) -%}${block}{%- endcall -%}`
    : `{{- ${macro.name}(${macro.params}) -}}`;

  return environment.renderString(
    `{%- from "${macro.path}" import ${macro.name} -%}
    ${callComponent}
    `,
    {},
  );
};
