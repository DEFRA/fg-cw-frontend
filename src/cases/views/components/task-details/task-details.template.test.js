import { describe, expect, test } from "vitest";
import { render } from "../../../../common/nunjucks/render.js";

describe("task-details", () => {
  test("renders", () => {
    const component = render("task-details", {
      caseId: "case-id",
      stageCode: "strage-id",
      taskGroupCode: "task-group-code",
      taskCode: "task-id",
      currentTask: {
        code: "task1",
        title: "Test Task",
        type: "boolean",
      },
    });

    expect(component).toMatchSnapshot();
  });
});
