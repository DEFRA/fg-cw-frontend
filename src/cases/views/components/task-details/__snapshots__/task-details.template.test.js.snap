// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`task-details > renders 1`] = `
"

<div class="govuk-grid-row">
  <div class="govuk-grid-column-two-thirds">
    <a href="/cases/case-id" class="govuk-back-link" data-testid="back-link">Back to task list</a>




    <h2
      class="govuk-heading-l govuk-!-margin-top-6"
      data-testid="task-details-heading"
    >
      Test Task
    </h2>

      
<form
  method="post"
  id="completeTaskForm"
  action="/cases/case-id/stages/strage-id/task-groups/task-group-id/tasks/task1/status"
>
  <div class="govuk-form-group">
    <div class="govuk-form-group">
  <div class="govuk-checkboxes" data-module="govuk-checkboxes">
    <div class="govuk-checkboxes__item">
      <input class="govuk-checkboxes__input" id="task-task1" name="isComplete" type="checkbox" value="true">
      <label class="govuk-label govuk-checkboxes__label" for="task-task1">
        Test Task
      </label>
    </div>
  </div>
</div>

  </div>


  <div class="govuk-button-group">
    <button
      type="submit"
      data-prevent-double-click="true"
      data-testid="save-and-continue-button"
      class="govuk-button"
      data-module="govuk-button"
      data-govuk-button-init=""
    >
      Save and continue
    </button>
  </div>
</form>

  </div>
</div>
"
`;
